nav2_world_model/src/world_model.cpp

```diff
@@ -12,67 +12,124 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-#include <iostream>
-#include <memory>
-#include <vector>
-#include <string>
 #include "nav2_world_model/world_model.hpp"
 
-using std::vector;
-using std::string;
+#include <iostream>
+#include <memory>
 
 namespace nav2_world_model
 {
 
-WorldModel::WorldModel(rclcpp::executor::Executor & executor, const string & name)
-: Node(name),
-  tfBuffer_(get_clock()),
-  tfListener_(tfBuffer_)
+WorldModel::WorldModel()
+: nav2_lifecycle::LifecycleNode("world_model")
+{
+  RCLCPP_INFO(get_logger(), "Creating");
+
+  costmap_ros_ = std::make_shared<nav2_costmap_2d::Costmap2DROS>("world_model_global_costmap");
+  costmap_thread_ = std::make_unique<std::thread>(
+    [](rclcpp_lifecycle::LifecycleNode::SharedPtr node)
+    {rclcpp::spin(node->get_node_base_interface());}, costmap_ros_);
+}
+
+WorldModel::~WorldModel()
+{
+  RCLCPP_INFO(get_logger(), "Destroying");
+  costmap_thread_->join();
+}
+
+nav2_lifecycle::CallbackReturn
+WorldModel::on_configure(const rclcpp_lifecycle::State & state)
 {
-  costmap_ros_ = std::make_shared<nav2_costmap_2d::Costmap2DROS>("global_costmap", tfBuffer_);
-  costmap_ = costmap_ros_->getCostmap();
-  executor.add_node(costmap_ros_);
+  RCLCPP_INFO(get_logger(), "Configuring");
+
+  costmap_ros_->on_configure(state);
 
   // Create a service that will use the callback function to handle requests.
-  costmapServer_ = create_service<nav2_msgs::srv::GetCostmap>("GetCostmap",
-      std::bind(&WorldModel::costmap_callback, this,
+  costmap_service_ = create_service<nav2_msgs::srv::GetCostmap>("GetCostmap",
+      std::bind(&WorldModel::costmap_service_callback, this,
       std::placeholders::_1, std::placeholders::_2, std::placeholders::_3));
+
+  return nav2_lifecycle::CallbackReturn::SUCCESS;
 }
 
-void WorldModel::costmap_callback(
+nav2_lifecycle::CallbackReturn
+WorldModel::on_activate(const rclcpp_lifecycle::State & state)
+{
+  RCLCPP_INFO(get_logger(), "Activating");
+
+  costmap_ros_->on_activate(state);
+
+  return nav2_lifecycle::CallbackReturn::SUCCESS;
+}
+
+nav2_lifecycle::CallbackReturn
+WorldModel::on_deactivate(const rclcpp_lifecycle::State & state)
+{
+  RCLCPP_INFO(get_logger(), "Deactivating");
+
+  costmap_ros_->on_deactivate(state);
+
+  return nav2_lifecycle::CallbackReturn::SUCCESS;
+}
+
+nav2_lifecycle::CallbackReturn
+WorldModel::on_cleanup(const rclcpp_lifecycle::State & state)
+{
+  RCLCPP_INFO(get_logger(), "Cleaning up");
+
+  costmap_ros_->on_cleanup(state);
+  costmap_service_.reset();
+
+  return nav2_lifecycle::CallbackReturn::SUCCESS;
+}
+
+nav2_lifecycle::CallbackReturn
+WorldModel::on_error(const rclcpp_lifecycle::State &)
```

Comment:
should we add `/*state*/` to be consistent with other PR's?

Reply:
Yes, will do.